### ISSUE TITLE
Smoke detector not discovered

### ISSUE THREAD

**Issue by domerich** [rospogrigio/localtuya#1078]
my WIFIDS10WT smoke detector is not discovered.

**Comment by remlei** [comment]
well its battery operated so... yeah. this is expected. same with other wifi based tuya device that is battery powered (eg, temperature, motion sensor etc...)

as always this is not a bug, its just that this device sleep a lot, and this device is only connected to wifi network for 10seconds or less, just enough to report something to the cloud.

that 10second is your main issue, after all you can really discover a device that is already disconnected to the wifi in the first place.

**Comment by leeyuentuen** [comment]
change the smoke detector with a Zigbee variant and the problem should be solved

**Comment by remlei** [comment]
Unfortunately as it seems, battery powered wifi sensor is just bad.

The only forgivable wifi based tuya sensor is the mmwave presence sensor since theres no way you will be powering this using a battery. 

Other else like PIR sensor, temperarure sensor, light sensor, door sensor, smoke sensor, water sensor etc... that are battery powered and uses wifi are pointless and basically a ewaste at this point.

Tl;dr just get a zwave/zigbee based sensor if your dealing with battery powered devices.

**Comment by domerich** [comment]
when you configure it right it works - partly. the devices all show unknown status, however in case of fire or low battery they will show that status. 

![image](https://user-images.githubusercontent.com/104911368/198546807-9449249a-0bb2-41ac-8f0b-cf0863863201.png)

**Comment by remlei** [comment]
Even if you managed to add it, it will be finicky to make it work with localtuya, after all these tuya device can on push notification to cloud, and you can only do polling in local connection which localtuya is not specifically designed to persistently reconnect to a device that rarely only get connected to wifi and this will probably flood the logs.

My best advice for these devices is just add this devices via official tuya integration.

**Comment by domerich** [comment]
> Even if you managed to add it, it will be finicky to make it work with localtuya, after all these tuya device can on push notification to cloud, and you can only do polling in local connection which localtuya is not specifically designed to persistently reconnect to a device that rarely only get connected to wifi and this will probably flood the logs.
> 
> My best advice for these devices is just add this devices via official tuya integration.

with official Tuya they don't work at all this is why I flashed them with tasmota

**Comment by RugbyAl** [comment]
My tuya battery powered smoke alarm is discovered by Tuya integration, and Local Tuya has it in the list when going to configure - add device, but it wont add. Checked "Get device specification attributes" in Tuya IoT, not surprisingly no DPS codes as a dormant battery powered device, can't be discovered / added to Local Tuya.  I guess this is not going to work with Local?

**Comment by SaschaKP** [comment]
Trigger the alarm manually and you can add them, but as others said, they can't be used locally since they only push to cloud, pretty pointless sensors with original firmware.
